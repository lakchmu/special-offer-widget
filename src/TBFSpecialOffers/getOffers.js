import request from './request';
import OffersStorage from './offersStorage';
import { API_END_POINT, API_METHOD_SPECIAL_OFFERS } from '../constants';

function getOffers(token, filters) {
  const url = API_END_POINT + API_METHOD_SPECIAL_OFFERS;
  return request(url, 'get', token)
    .then(function(response) {
      return response.json();
    })
    .then(json => {
      return new OffersStorage(json.special_offers);
    });
}

export default getOffers;