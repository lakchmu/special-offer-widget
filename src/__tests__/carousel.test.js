/* global jest, describe, it, expect */

import getOffers from '../TBFSpecialOffers/getOffers';
import OffersView from '../TBFSpecialOffers/offersView';

jest.mock('../TBFSpecialOffers/request');
jest.mock('../index.css');

function setup() {
  const data = {
    token: 'KO96pMbbIBTQPl7vnoJMXmIrEl2cYoUZS83QlL0Abu0',
    filters: {},
  };
  return getOffers(data.token, data.filters);
}

describe('Testing Carousel class', () => {
  it('Testing render function', () => {
    document.body.innerHTML = '<div class="special-offers"></div>';
    const root = document.querySelector('.special-offers');
    const options = { view: 'carousel' };
    expect.assertions(1);
    return setup().then((offersStorage) => {
      const offersView = new OffersView('.special-offers', offersStorage, options);
      expect(root).toMatchSnapshot();
    });
  });

  it('Testing arrow event listener', () => {
    document.body.innerHTML = '<div class="special-offers"></div>';
    const root = document.querySelector('.special-offers');
    const options = { view: 'carousel' };
    expect.assertions(4);
    return setup().then((offersStorage) => {
      const offersView = new OffersView('.special-offers', offersStorage, options);
      const leftArrow = document.querySelector('.tbf-so-arrow[data-direction="to-left"]');
      const rightArrow = document.querySelector('.tbf-so-arrow[data-direction="to-right"]');
      const indicators = document.querySelectorAll('.tbf-so-carousel-indicator');
      rightArrow.click();
      expect(document.querySelectorAll('.tbf-so-carousel-indicator.active').length).toBe(1);
      expect(indicators[1].classList.contains('active')).toBeTruthy();
      leftArrow.click();
      expect(document.querySelectorAll('.tbf-so-carousel-indicator.active').length).toBe(1);
      expect(indicators[0].classList.contains('active')).toBeTruthy();
    });
  });

  it('Testing indicator event listener', () => {
    document.body.innerHTML = '<div class="special-offers"></div>';
    const root = document.querySelector('.special-offers');
    const options = { view: 'carousel' };
    expect.assertions(4);
    return setup().then((offersStorage) => {
      const offersView = new OffersView('.special-offers', offersStorage, options);
      const indicators = document.querySelectorAll('.tbf-so-carousel-indicator');
      indicators[1].click();
      expect(indicators[1].classList.contains('active')).toBeTruthy();
      expect(document.querySelectorAll('.tbf-so-carousel-indicator.active').length).toBe(1);
      indicators[0].click();
      expect(indicators[0].classList.contains('active')).toBeTruthy();
      expect(document.querySelectorAll('.tbf-so-carousel-indicator.active').length).toBe(1);
    });
  });
});
