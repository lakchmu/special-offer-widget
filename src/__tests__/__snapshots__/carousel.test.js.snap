// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Carousel class Testing render function 1`] = `
<div
  class="special-offers tbf-so-special-offers tbf-so-carousel"
>
  
    
  <div
    class="tbf-so-carousel-content"
  >
    
      
    <div
      class="tbf-so-offer-list"
    >
      
        
  
      <div
        class="tbf-so-offer"
      >
        
    
        <div
          class="tbf-so-offer__content"
        >
          
      
          <div
            class="tbf-so-offer__content-image"
          >
            
        
            <img
              class="tbf-so-offer__image "
              src="https://s3.amazonaws.com/buuqit-dev/special_offers/images/000/000/009/public/Room2.jpg?1515606281"
            />
            
      
          </div>
          
      
          <div
            class="tbf-so-offer__content-text"
          >
            
        
            <div
              class="tbf-so-offer__title"
            >
              Last Minute Booking
            </div>
            
        
            <div
              class="tbf-so-offer__wrapper-description"
            >
              
          
              <div
                class="tbf-so-offer__short-description"
              >
                Last minute rate, Non-Refundable
              </div>
              
          
              <div
                class="tbf-so-offer__description"
              >
                Last minute booking discount, full pre payment required on time of booking. Please note this rate is non-refundable.
              </div>
              
        
            </div>
            
        
            <div
              class="tbf-so-offer__footer"
            >
              
          
              <span
                class="tbf-so-offer__dates"
              >
                
            
                <i
                  class="tbf-so-offer__icon-calendar"
                />
                Jun 9 - Aug 31
          
              </span>
              
          
              <span
                class="tbf-so-offer__control-links"
              >
                
            
                <button
                  class="tbf-so-offer__more-link"
                >
                  More
                </button>
                
            
                <a
                  class="tbf-so-offer__booking-link"
                  href="https://app.thebookingfactory.com/Spilman-Hotel/book/__special_offer_9"
                >
                  Book now
                </a>
                
          
              </span>
              
        
            </div>
            
      
          </div>
          
    
        </div>
        
  
      </div>
      


  
      <div
        class="tbf-so-offer"
      >
        
    
        <div
          class="tbf-so-offer__content"
        >
          
      
          <div
            class="tbf-so-offer__content-image"
          >
            
        
            <img
              class="tbf-so-offer__image "
              src="https://s3.amazonaws.com/buuqit-dev/special_offers/images/000/000/006/public/Room7.jpg?1512864349"
            />
            
      
          </div>
          
      
          <div
            class="tbf-so-offer__content-text"
          >
            
        
            <div
              class="tbf-so-offer__title"
            >
              Stay 3 Nights Save 15%
            </div>
            
        
            <div
              class="tbf-so-offer__wrapper-description"
            >
              
          
              <div
                class="tbf-so-offer__short-description"
              >
                Stay 3 Nights Save 15%
              </div>
              
          
              <div
                class="tbf-so-offer__description"
              >
                Stay for 3 nights or more and save 15%. Save a little by staying with us a bit longer
              </div>
              
        
            </div>
            
        
            <div
              class="tbf-so-offer__footer"
            >
              
          
              <span
                class="tbf-so-offer__dates"
              >
                
            
                <i
                  class="tbf-so-offer__icon-calendar"
                />
                Jun 4 - Aug 31
          
              </span>
              
          
              <span
                class="tbf-so-offer__control-links"
              >
                
            
                <button
                  class="tbf-so-offer__more-link"
                >
                  More
                </button>
                
            
                <a
                  class="tbf-so-offer__booking-link"
                  href="https://app.thebookingfactory.com/Spilman-Hotel/book/__special_offer_6"
                >
                  Book now
                </a>
                
          
              </span>
              
        
            </div>
            
      
          </div>
          
    
        </div>
        
  
      </div>
      

      
    </div>
    
    
  </div>
  
    
  <div
    class="tbf-so-carousel-indicators"
  >
    
      
    <div
      class="tbf-so-arrow"
      data-direction="to-left"
    />
    
      
    <div
      class="tbf-so-carousel-indicator active"
      data-number-offer="0"
    />
    

    <div
      class="tbf-so-carousel-indicator"
      data-number-offer="1"
    />
    
      
    <div
      class="tbf-so-arrow"
      data-direction="to-right"
    />
    
    
  </div>
  
    
</div>
`;
